src/__tests__/pipeline.test.js

```diff
@@ -9,9 +9,21 @@ function isEven (x) {
   return (x % 2) === 0
 }
 
+const add = (a, b) => a + b
+const mul = (a, b) => a * b
+
 describe('pipeline', function () {
   it('works', function () {
     const iter = pipeline([1, 2, 3, 4], map(bump), filter(isEven))
     expect(Array.from(iter)).toEqual([4, 6])
   })
+
+  it('can create a complex data processing pipeline', function () {
+    expect(Array.from(pipeline(
+      [3, 3],
+      fork,
+      slice(2),
+      merge(apply).partial([add, mul])
```

Comment:
partial?
* I thought we discarded the idea of partial method
* I don't see it implemented, how can this pass the test ?